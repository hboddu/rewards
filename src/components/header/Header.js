import React from 'react';
import { Box, Typography } from '@mui/material';
import  HeaderStyles from './HeaderStyles';

const Header = () => {

    const { classes } = HeaderStyles();

    return (
        <Box>
            <Typography className={classes.headline} variant='h1' data-testid="header_text">Quarterly Rewards Statement</Typography>
        </Box>
    )
}

export default Header;