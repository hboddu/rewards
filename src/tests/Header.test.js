import { render, screen, cleanup } from '@testing-library/react';
import Header from '../components/header/Header';
import renderer from 'react-test-renderer';

afterEach(() => {
    cleanup(); 
})

describe("Header Component" ,() => {

    test('renders Table Heading', () => {
        render(<Header />);
        const headerElement = screen.getByTestId("header_text");
        expect(headerElement).toBeInTheDocument();
        expect(headerElement).toBeInTheDocument();
    });

    test('Table Heading Snapshot', () => {
        const HeaderComponent = renderer.create(<Header />);
        let tree = HeaderComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });

});