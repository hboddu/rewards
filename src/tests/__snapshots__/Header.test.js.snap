// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Table Heading Snapshot 1`] = `
<div
  className="MuiBox-root css-0"
>
  <h1
    className="MuiTypography-root MuiTypography-h1 css-18wfj3a-MuiTypography-root-Header-headline"
    data-testid="header_text"
  >
    Quarterly Rewards Statement
  </h1>
</div>
`;
